docs/synapse/userguides/background.rst

```diff
@@ -0,0 +1,208 @@
+
+.. _bkd-directed-graphs:
+
+Directed Graphs
+---------------
+
+A **directed graph** is a graph where the edges have a direction associated with them. In other words, the
+relationship represented by the edge is one-way. Where an edge in an undirected graph is often represented by
+a straight line, an edge in a directed graph is represented by an arrow.
+
+|directedgraph|
+
+**Examples**
+
+**Cities and Roads.** In our cities-and-roads example, the graph would be a directed graph if the roads were
+all one-way streets: in this case you can use a particular road to go from City A to City B, but not from City
+B to City A.
+
+**Social Networks.** Social networks that support a "follows" relationship (such as Twitter) can be represented
+as directed graphs. Each person is still a node, but the "follows" relationship is one way – I can "follow" you,
+but you don’t have to follow me. If you choose to follow me, that would be a second, independent one-way edge
+in the opposite direction. (This is also a simplification but works for a basic illustration.)
+
+**Other Examples.** Many other types of data can be represented with nodes and directed edges.  For example, in
+information security you can represent data and relationships such as:
+
+``malware_file --(performed DNS lookup for)--> domain``
+
+or
+
+``domain --(resolves to)--> ip_address``
+
+In these examples, files, domains and IP addresses are nodes and "performed DNS lookup for" and "resolves to"
+are edges (relationships). The edges are directed because a malware binary can contain programming to resolve
+a domain name, but a domain can’t "perform a lookup" for a malware binary; the relationship (edge) is one-way.
+
+In addition to nodes and edges, some directed graph implementations may allow labeling or tagging of nodes and
+edges with additional information. These tags can act as metadata for various purposes, such as to create
+analytically relevant groups of objects.
+
+Many tools exist to visually represent various types of data in a directed graph format.
+
+.. _bkd-graph-analysis:
+
+Analysis with Graphs
+--------------------
+
+When working with graphs and directed graphs, analysts typically select (or lift) objects (nodes) and
+navigate the graph by traversing the edges (relationships) that connect those nodes. A key limitation to
+this approach is that all relationships (edges) between objects must be explicitly defined. You must know
+all of the relationships that you want to represent in advance, which makes the discovery of novel relationships
+among the data extremely difficult.
+
+.. _bkd-hypergraphs:
+
+Hypergraphs
+-----------
+
+A **hypergraph** is a generalization of a graph in which an edge can join any number of nodes. Because an edge
+is no longer limited to joining exactly two nodes, edges in a hypergraph are often called **hyperedges.** If a
+directed graph where edges join exactly two nodes is two-dimensional, then a hypergraph where a hyperedge can
```

Comment:
We may want to strike this sentence to prevent debate.

Reply:
Still seeing reference to `n-dimentional`

Reply:
Has been fixed - comment relates to an outdated branch.